redis_unbounded_segments_test.go

```diff
@@ -0,0 +1,54 @@
+package ldredis
+
+import (
+	"fmt"
+	"testing"
+
+	"gopkg.in/launchdarkly/go-server-sdk.v5/interfaces"
+	"gopkg.in/launchdarkly/go-server-sdk.v5/testhelpers/storetest"
+
+	r "github.com/gomodule/redigo/redis"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnboundedSegmentStore(t *testing.T) {
+	client, err := r.DialURL(redisURL)
+	require.NoError(t, err)
+	defer client.Close()
+
+	setTestMetadata := func(prefix string, metadata interfaces.UnboundedSegmentStoreMetadata) error {
+		if prefix == "" {
+			prefix = DefaultPrefix
+		}
+		_, err := client.Do("SET", unboundedSegmentsSyncTimeKey(prefix), fmt.Sprintf("%d", metadata.LastUpToDate))
+		return err
+	}
+
+	setTestKeys := func(prefix string, userHashKey string, included []string, excluded []string) error {
+		if prefix == "" {
+			prefix = DefaultPrefix
+		}
+		for _, inc := range included {
+			_, err := client.Do("SADD", unboundedSegmentsIncludeKey(prefix, userHashKey), inc)
+			if err != nil {
+				return err
+			}
+		}
+		for _, exc := range excluded {
+			_, err := client.Do("SADD", unboundedSegmentsExcludeKey(prefix, userHashKey), exc)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	storetest.NewUnboundedSegmentStoreTestSuite(
```

Comment:
The `UnboundedSegmentStoreTestSuite` is implemented in the `testhelpers/storetest` package of `go-server-sdk` in private PR 288. The reason it's configured here in a somewhat different way than `PersistentDataStoreTestSuite`—i.e. we have to pass in functions that the test code will call to put data in the database—is that the `UnboundedSegmentStore` interface itself doesn't have any methods for updating data, as `PersistentDataStore` does.